bots/example_bots/gridbot.py

```diff
@@ -232,6 +197,56 @@ def cancel_all_active_orders(self) -> None:
     # helper methods
     ######################################################################
 
+    def _amount_in_market(self, side: OrderSide) -> Decimal:
+        active_orders = list(filter(lambda order: order.order_side == side, self.active_limit_orders.values()))
```

Comment:
at first i thought that once we have the `view functions` that expose to a maker their individual offers, attempting to use chain state could be a more sound approach. however, that does introduce a redundant call (assuming the bot is properly tracking its book) on a function that will routinely be called. somewhere down the line adding a redundant call somewhere to sanity check against chain state could be very nice

Reply:
I think adding this redundant call definitely makes sense however doing it in a separate thread so that it does not happen on the main execution path of the bot but rather as a sanity check on a set period.

Then it has the benefit of ensuring the tracked orders are correct while also not introducing additional latency